tests/contrib/data/url2purl.json

```diff
@@ -139,5 +139,33 @@
 
   "https://crates.io/api/v1/crates/rand/0.7.2/download": "pkg:cargo/rand@0.7.2",
   "https://crates.io/api/v1/crates/clap/2.33.0/download" : "pkg:cargo/clap@2.33.0",
-  "https://crates.io/api/v1/crates/structopt/0.3.11/download" : "pkg:cargo/structopt@0.3.11"
+  "https://crates.io/api/v1/crates/structopt/0.3.11/download" : "pkg:cargo/structopt@0.3.11",
+
+  "https://github.com/TG1999/fetchcode/tree/documentation/fetchcode" : "pkg:github/tg1999/fetchcode@documentation#fetchcode",
+  "https://github.com/nexB/scancode-toolkit/tree/develop/plugins/scancode-ctags-macosx_10_9_intel" : "pkg:github/nexb/scancode-toolkit@develop#plugins/scancode-ctags-macosx_10_9_intel",
+  "https://github.com/package-url/packageurl-js/tree/master/test/data" : "pkg:github/package-url/packageurl-js@master#test/data",
+  "https://github.com/package-url/packageurl-js/tree/master" : "pkg:github/package-url/packageurl-js@master",
+  "https://github.com/package-url/packageurl-js" : "pkg:github/package-url/packageurl-js",
+  "https://github.com/TG1999":null,
+  "https://github.com/TG1999/fetchcode/tree":null,
+  "https://github.com/TG1999/fetchcode/master": null,
+  "https://github.com/TG1999/fetchcode/fetchcode":null,
```

Comment:
Why would  all these be null? No null except for "https://github.com/TG1999":null,

Reply:
These URLs doesn't point anywhere, they all return 404, that's why I return null, so what say :D

Reply:
OK, they may be 404, but there is still a valid purl we can get from these. We should get it IMHO, not a null... I would be interested to get someone else take too.